service/controller/resource/app/desired.go

```diff
@@ -421,6 +427,61 @@ func newUserConfig(cr apiv1beta1.Cluster, appSpec key.AppSpec, configMaps map[st
 	return userConfig
 }
 
+func (r *Resource) getAppExtraConfigs(ctx context.Context, cr apiv1beta1.Cluster, appSpec key.AppSpec, configMaps map[string]corev1.ConfigMap, secrets map[string]corev1.Secret) []g8sv1alpha1.AppExtraConfig {
+	var err error
+	var ret []g8sv1alpha1.AppExtraConfig
+
+	for name, cm := range configMaps {
+		if strings.HasPrefix(name, appSpec.App) && name != key.AppUserConfigMapName(appSpec) {
+			priority := DefaultAppExtraConfigPriority
+			{
+				priorityStr, found := cm.Annotations[annotation.AppConfigPriority]
```

Comment:
There are some constrains on what the priority level can be set to. I am wondering what happens when the annotation contains an invalid value. The API will reject it and we might end up in a loop.

There are some helper methods and constants defined here that can help adding a basic validation here, please:
- https://github.com/giantswarm/app/blob/master/pkg/values/extra_configs.go
- https://github.com/giantswarm/app/blob/master/pkg/values/extra_configs.go

Reply:
Pushed an update for this too.

Reply:
Working a few simple test cases for the validation, but could you please add some tests for the extra configs if it is not too much of a trouble? LGTM otherwise.

Reply:
thanks. LGTM. Please remove your change request since you made the changes yourself

Reply:
Added the unit tests for the conversion + validation of the priority read from the annotation.